lib/nerves_network/wifi_manager.ex

```diff
@@ -280,6 +280,24 @@ defmodule Nerves.Network.WiFiManager do
     %Nerves.Network.WiFiManager{state | wpa_pid: pid}
   end
 
+  defp parse_settings(settings) when is_list(settings) do
+    settings
+    |> Map.new
+    |> parse_settings
+  end
+
+  defp parse_settings(settings = %{ key_mgmt: key_mgmt }) when is_binary(key_mgmt) do
+    %{ settings | key_mgmt: String.to_atom(key_mgmt) }
+    |> parse_settings
+  end
+
+  defp parse_settings(settings = %{ key_mgmt: :NONE, psk: _psk }) do
+    Map.delete(settings, :psk)
```

Comment:
Adding to comments #38. 

I actually like this one. BUT - I believe it should be expanded. Not in this PR per say (unless you feel so inclined) I think this function should enforce the configuration api.